core/src/cg/group4/rewards/LocalCollectionReader.java

```diff
@@ -0,0 +1,103 @@
+package cg.group4.rewards;
+
+import cg.group4.exceptions.LocalStoreUnavailableException;
+import com.badlogic.gdx.Gdx;
+import com.badlogic.gdx.files.FileHandle;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectStreamException;
+
+/**
+ * Reads the collection data from the local storage.
+ * Reason for not using the internal storage: is read only.
+ */
+public class LocalCollectionReader implements CollectionReader {
+
+    /**
+     * Debug tag containing the simple class name. Used for debugging purposes.
+     */
+    protected String cTag = this.getClass().getSimpleName();
+
+    /**
+     * Path to the file which contains the collection data to be deserialized.
+     */
+    protected final String cLocalFile;
+
+    /**
+     * Initialization of the local collection reader.
+     */
+    public LocalCollectionReader() {
+        cLocalFile = "starfish.save";
+    }
+
+    /**
+     * Initialization of the local collection reader.
+     * @param saveFileName name of the file from which the collection game save will be read.
+     */
+    public LocalCollectionReader(final String saveFileName) {
+        cLocalFile = saveFileName;
+    }
+
+    @Override
+    public Collection read() {
```

Comment:
wouldn't it make more sense if it was given the file to read, perhaps in the form of a collection id that the remote reader could use to retrieve that collection from the server.